### DISCUSSION TITLE
Grafana Telegram Alerts: Automatically migrate to new chat ID

### DISCUSSION THREAD

**Discussion by Miroka96** [grafana/grafana#38117] [Alerting]
Hello!

My Grafana Telegram Alerts suddenly stopped working and created the following log lines:
```
lvl=eror msg="Failed to send webhook" logger=alerting.notifier.telegram error="Webhook response status 400 Bad Request" webhook=Telegram
lvl=eror msg="failed to send notification" logger=alerting.notifier uid= error="Webhook response status 400 Bad Request"
lvl=eror msg="failed to send notification" logger=alerting.notifier uid= error="Webhook response status 400 Bad Request"
lvl=eror msg="Failed to send alert notifications" logger=context userId=1 orgId=1 uname=admin error="Webhook response status 400 Bad Request" remote_addr=1.1.1.1
```
Later, I found out that this was caused by myself when I modified the Telegram group settings in order to enable new group members to see previous messages (made history visible to new members). Under the hood, Telegram turned my group into a super group, which has a different group ID. Requests to Telegrams getUpdates API (https://api.telegram.org/botDDDDDDDDDDDDDDD:XXXXXXXXXXX/getUpdates) then returned a new field called `migrate_to_chat_id` where I could find the new chat ID. Entering this new chat ID in the Notification Channel's chat ID field resolved the issue.

**What would you like to be added**:

It would be nice, if Grafana automatically detects and applies ID migrations. Otherwise, it would be nice to have a piece of documentation for this or have at least debug messages containing the real request to ease debugging of such foreign API issues.

**Why is this needed**:

It took me much too long to find out where this generic "Webhook response status 400 Bad Request" error was coming from.

Cheers,
Miroka

P.S.: Even if you instantly close this issue, I hope that someone finds the solution helpful.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by mxchist** [comment]
I have encountered the same behavior of Telegram alert when my group implicitly became a super group
